### PR TITLE
Update fairscale version

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?
Part of #11207
The [change](https://github.com/facebookresearch/fairscale/pull/885) in fairscale side released in 0.4.5, upgrade lightning dependence version to 0.4.5 too

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
Hi @four4fish! I had #11478 already open but its fine, I'll close mine 😃

This is currently blocked by #11566

**Comment by four4fish** [comment]
> Hi @four4fish! I had #11478 already open but its fine, I'll close mine 😃
> 
> This is currently blocked by #11566

@carmocca sorry i didn't see you already have one open, that works, thank you! :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -222,6 +222,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Changed `MisconfigurationException` to `ModuleNotFoundError` when `rich` isn't available ([#11360](https://github.com/PyTorchLightning/pytorch-lightning/pull/11360))
 
 
+- Updated `fairscale` version ([#11567](https://github.com/PyTorchLightning/pytorch-lightning/pull/11567))
+
+
```
Comment (akihironitta) [inline CHANGELOG.md:227]: We don't need an entry for a change in CI, right?
```suggestion
```

**Review by akihironitta — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: .azure-pipelines/gpu-tests.yml [hunk 2]
```diff
@@ -50,7 +50,7 @@ jobs:
 
     - bash: |
         python -c "fname = 'requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'horovod' not in line] ; open(fname, 'w').writelines(lines)"
-        pip install fairscale==0.4.0
+        pip install fairscale==0.4.5
```
Comment (carmocca) [inline .azure-pipelines/gpu-tests.yml:53]: There's a failing test:

tests/strategies/test_ddp_fully_sharded_with_full_state_dict.py::test_fully_sharded_strategy_checkpoint

Comment (four4fish) [reply .azure-pipelines/gpu-tests.yml:53]: Thanks @carmocca 
It caused by a fairscale bug fix  "[Fixed a corner case of FSDP init order and losing one of the flags](https://github.com/facebookresearch/fairscale/commit/7c2c3e0046eb3a1b1c98fd24f61fb09f90f6817f)" in version 0.4.4
Our tests needs to be updated, I will update them later

Comment (four4fish) [reply .azure-pipelines/gpu-tests.yml:53]: The ci gpu is broken, and I couldn't reproduce this on my machine, so will have to wait

Comment (akihironitta) [reply .azure-pipelines/gpu-tests.yml:54]: @four4fish The CI should be back now. Would you mind having a look?

File: .azure-pipelines/gpu-tests.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ jobs:
...
```
Comment (akihironitta) [inline .azure-pipelines/gpu-tests.yml:54]: Should we pin its version at the latest one `0.4.6`, or should we simply unpin it to always use the latest version? (I'm curious why we need to pin its version in the first place :eyes:)

Comment (four4fish) [reply .azure-pipelines/gpu-tests.yml:54]: I think there is compatibility issue. The shard_on_root behavior has changed, I haven't figure out a way to fix the test yet
